docs/cumulusci_flow.rst

```diff
@@ -0,0 +1,273 @@
+   * ``feature/large-feature``
+   * ``feature/large-feature__section1``
+   * ``feature/large-feature__section1__work-item1``
+   * ``feature/large-feature__section1__work-item2``
+   * ``feature/large-feature__section2``
+   * ``feature/large-feature__section2__work-item1``
+
+In this scenario, a commit to the ``main`` branch triggers the ``github_automerge_main`` task to run and will automerge that commit into ``feature/large-feature``.
+This triggers a build to run against ``feature/large-feature``, and assuming the build passes, runs the ``github_automerge_feature`` task.
+This task detects two child branches of ``feature/large-feature``; ``feature/large_feature__section1`` and ``feature/large-feature__section2``.
+The task automerges the commit from the parent, into the child branches, and builds begin to run against those branches.
+If the build for ``feature/large-feature__section1`` fails; then it would not trigger ``github_automerge_feature`` against it.
+This means that despite ``feature/large-feature__section1`` having two child branches, they would not receive automerges.
+
+
+
+Release to (Future) Release Merges
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Because release branches are so long-lived, and so much work goes into them, their diffs can get quite large.
+This means headaches are inevitable the day after a major release, and you need to pull down all of the changes from the new release into the next release branch (which has likely been in development for months already).
+To alleviate this pain point, CumulusCI can ensure that all release branches propagate commits they receive to other existing release branches that correspond to future releases.
+
+Consider the following branches in a GitHub repository:
+
+   * ``main`` - Source of Truth for Production
+   * ``feature/002`` - The next major production release
+   * ``feature/002__feature1`` - A single feature associated with release ``002``
+   * ``feature/002__large_feature`` - A large feature associated with release ``002``
+   * ``feature/002__large_feature__child1`` - First chunk of work for the large feature
+   * ``feature/002__large_feature__child2`` - Second chunk of work for the large feature
+   * ``feature/003`` - The release that comes after ``002``
+   * ``feature/003__feature1`` - A single feature associated with release ``003``
+
+In this scenario, CumulusCI ensures that when ``feature/002`` receives a commit, that that commit is also merged into ``feature/003``.
+This kicks off tests in our CI system and ensures that functionality going into ``feature/002`` doesn't break work being done for future releases.
+Once those tests pass, the commit on ``feature/003`` is merged to ``feature/003__feature1`` because they adhere to the parent/child naming convention described above.
+Commits **never** propagate in the opposite direction. (A commit to ``feature/002`` would never be merged to ``feature/001`` if it was an existing branch in the GitHub repository).
```

Comment:
Just calling out that I really like these scenarios and merge diagrams. I think this is extremely helpful for understanding a much more complex automerge landscape.